src/ngraph/frontend/onnxifi/tensor.hpp

```diff
@@ -0,0 +1,141 @@
+//*****************************************************************************
+// Copyright 2017-2018 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#pragma once
+
+#include <memory>
+
+#include <onnxifi.h>
+
+#include "ngraph/runtime/backend.hpp"
+#include "ngraph/runtime/tensor.hpp"
+
+#include "exceptions.hpp"
+#include "span.hpp"
+
+namespace ngraph
+{
+    namespace onnxifi
+    {
+        /// \brief Wrapper to onnxTensorDescriptorV1 class
+        class Tensor
+        {
+        public:
+            Tensor(const Tensor&) = default;
+            Tensor& operator=(const Tensor&) = default;
+
+            Tensor(Tensor&&) = default;
+            Tensor& operator=(Tensor&&) = default;
+
+            Tensor() = delete;
+            virtual ~Tensor() = default;
+
+            explicit Tensor(const ::onnxTensorDescriptorV1& tensor)
+                : m_tensor{&tensor}
+            {
+                if (tensor.tag != ONNXIFI_TAG_TENSOR_DESCRIPTOR_V1)
```

Comment:
Please keep your headers free of implementations.

Reply:
done